maker/src/routes.rs

```diff
@@ -124,6 +126,12 @@ pub async fn put_offer_params(
     maker: &State<Maker>,
     _auth: Authenticated,
 ) -> Result<(), HttpApiProblem> {
+    // temporary until the automation can deal with this
+    let mut leverage_choices = offer_params.leverage_choices.clone();
+    if leverage_choices.is_empty() {
+        leverage_choices.push(Leverage::TWO)
+    }
+
```

Comment:
🔧 I don't think this works if I understand your intentions.

This is a breaking change to the automation i.e. the API expects a `leverage_choices` field now. One way to fix this would be to annotate the field as `#[serde(default)]`. That way a missing `leverage_choices` is replaced with the default value for `Vec<Leverage>` i.e. an empty vector.

Even better would be to use https://serde.rs/field-attrs.html#default--path, and you can provide a function which already returns a `vec![Leverage::TWO]`.

Reply:
Works without any problem at the moment 🙃

Reply:
You can run this against an old `automaker` without trouble? I'm very surprised.

I still think it would be cleaner to leverage `#[serde(default = "path")]`.

Reply:
Done in: [`6ad195d` (#1965)](https://github.com/itchysats/itchysats/pull/1965/commits/6ad195d0f80e4a0220786cea4fb585ab381dfb38)